### PR TITLE
feat: Adds the ability to publish an article

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

Adds the ability to publish, Unpublish and Draft articles through the article header.

Fixes https://github.com/chatwoot/product/issues/555

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

**Screencast**
https://www.loom.com/share/d95f7e70f7ed4527b81646b88247c6be

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by muhsin-k** [comment]
@iamsivin https://www.loom.com/share/c356789fe5de44aca68b427d427302fb

File: app/javascript/dashboard/routes/dashboard/helpcenter/constants.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export default {
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/helpcenter/constants.js:1]: Please use the main constants file https://github.com/chatwoot/chatwoot/blob/develop/app/javascript/dashboard/constants.js
```
import wootConstants from 'dashboard/constants';
const {
    ARTICLE_STATUS_TYPES
} = wootConstants;
```

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/Header/EditArticleHeader.vue [hunk 2]
```diff
@@ -111,6 +167,39 @@ export default {
...
+        if (status === this.ARTICLE_STATUS_TYPE.DRAFT) {
+          this.alertMessage = this.$t('HELP_CENTER.DRAFT_ARTICLE.API.SUCCESS');
+        } else if (status === this.ARTICLE_STATUS_TYPE.PUBLISH) {
+          this.alertMessage = this.$t(
+            'HELP_CENTER.PUBLISH_ARTICLE.API.SUCCESS'
+          );
+        } else if (status === this.ARTICLE_STATUS_TYPE.ARCHIVE) {
+          this.alertMessage = this.$t(
+            'HELP_CENTER.ARCHIVE_ARTICLE.API.SUCCESS'
+          );
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/Header/EditArticleHeader.vue:186]: Move to a method

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/Header/EditArticleHeader.vue [hunk 3]
```diff
@@ -111,6 +167,39 @@ export default {
...
+        }
+        this.closeActionsDropdown();
+      } catch (error) {
+        const errorMessage = '';
+        if (status === this.ARTICLE_STATUS_TYPE.DRAFT) {
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/Header/EditArticleHeader.vue:191]: Create method for this also

**Review by muhsin-k — COMMENTED** [review]
@iamsivin Please see the comments

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/Header/EditArticleHeader.vue [hunk 4]
```diff
@@ -100,6 +150,12 @@ export default {
         ? this.$t('HELP_CENTER.EDIT_HEADER.SAVING')
         : this.$t('HELP_CENTER.EDIT_HEADER.SAVED');
     },
+    articleId() {
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/Header/EditArticleHeader.vue:153]: Rename to articleSlug

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/Header/EditArticleHeader.vue [hunk 5]
```diff
@@ -100,6 +150,12 @@ export default {
...
+      return this.$route.params.articleSlug;
+    },
+    selectedPortalSlug() {
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/Header/EditArticleHeader.vue:156]: Use current for referring to selected portal

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/Header/EditArticleHeader.vue [hunk 6]
```diff
@@ -161,4 +259,7 @@ export default {
     }
   }
 }
+::v-deep.button-group .button {
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/Header/EditArticleHeader.vue:262]: We could remove these overrides?

**Review by nithindavid — CHANGES_REQUESTED** [review]

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/Header/EditArticleHeader.vue (hunk 6 above)
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/helpcenter/components/Header/EditArticleHeader.vue:262]: If we move this font size will be small. For small-size buttons, the font size is mini. In the button-group class the font size will be small.
image

Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/helpcenter/components/Header/EditArticleHeader.vue:264]: @nithindavid

**Comment by iamsivin** [comment]
> @iamsivin https://www.loom.com/share/c356789fe5de44aca68b427d427302fb

Done https://github.com/chatwoot/chatwoot/pull/5365/commits/ec84d1eee6bb6175403dda03717c397dd574877c

**Comment by muhsin-k** [comment]
@iamsivin @nithindavid How about showing "Unpublish article" instead of disabled "Publish article"?
Screenshot 2022-09-01 at 10 47 05 PM

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/Header/EditArticleHeader.vue (hunk 6 above)
Comment (nithindavid) [reply app/javascript/dashboard/routes/dashboard/helpcenter/components/Header/EditArticleHeader.vue:285]: Let it be like this, This should be fixed on the global button group classes

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/Header/EditArticleHeader.vue [hunk 7]
```diff
@@ -166,4 +282,7 @@ export default {
...
+  font-size: var(--font-size-mini);
+}
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/Header/EditArticleHeader.vue:287]:
```suggestion
```

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/Header/EditArticleHeader.vue (hunk 6 above)
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/helpcenter/components/Header/EditArticleHeader.vue]: Done https://github.com/chatwoot/chatwoot/pull/5365/commits/7492e0c52571b57c567e45aeface7cb38726699e

**Comment by nithindavid** [comment]
> 

There's a chance people might click it accidentally thinking it's to publish. Might cause confusion as it's the CTA.

**Comment by nithindavid** [comment]
@iamsivin The draft article is not needed as we have auto save. 

 Can we change the dropdown buttons texts to
1. Move to archived

**Comment by iamsivin** [comment]
> @iamsivin The draft article is not needed as we have auto save.
> 
> Can we change the dropdown buttons texts to
> 
> 1. Move to archived

@nithindavid Updated https://github.com/chatwoot/chatwoot/pull/5365/commits/ddb53355bc0405df63da85aed3beb2c61865513e


https://user-images.githubusercontent.com/64252451/188068746-90185343-5cd2-4ebe-a31c-2a4641519fdd.mov

**Review by nithindavid — APPROVED** [review]

**Pull request merged**